dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/processor/TaskExecuteResponseProcessor.java

```diff
@@ -62,13 +62,13 @@ public void process(Channel channel, Command command) {
                 channel,
                 taskExecuteResultMessage.getMessageSenderAddress());
         try {
-            LoggerUtils.setWorkflowAndTaskInstanceIDMDC(taskResultEvent.getProcessInstanceId(),
+            LogUtils.setWorkflowAndTaskInstanceIDMDC(taskResultEvent.getProcessInstanceId(),
                     taskResultEvent.getTaskInstanceId());
             log.info("Received task execute result, event: {}", taskResultEvent);
 
             taskEventService.addEvent(taskResultEvent);
         } finally {
-            LoggerUtils.removeWorkflowAndTaskInstanceIdMDC();
+            LogUtils.removeWorkflowAndTaskInstanceIdMDC();
```

Comment:
A suggestion that you can polish in this PR or in another PR: you can see currently we have many codes like

```java
try {
  LogUtils.setXxxx
} finally {
  LogUtils.removeXxx
}
```

This is kind of boilerplate codes, and the worse thing is if contributors forgot to call `LogUtils.removeXxxx` in the `finally` block, the logs may mess up.

What do you think we make the methods `LogUtils.setXxx` returns an `AutoCloseable`, and wrap the `LogUtils.removeXxx` inside the `AutoCloseable`, so that:

- We can use the try-with-resources like `try (final AutoCloseable ignored = LogUtils.setXxxx()) { ... }`
- If we forgot to call the `LogUtils.removeXxx` or wrap it in try-with-resources block, IDEs can warn us about that.

My suggestion is to add a class such as 


```java
    @RequiredArgsConstructor
    public class MDContext implements AutoCloseable {
        private final Runnable closeAction;

        @Override
        public void close() {
            closeAction.run();
        }
    }
```

And make `LogUtils.setXxx` return a `MDContext`, like this:

```java
    public static MDContext setWorkflowAndTaskInstanceIDMDC(Integer workflowInstanceId, Integer taskInstanceId) {
        setWorkflowInstanceIdMDC(workflowInstanceId);
        setTaskInstanceIdMDC(taskInstanceId);
        return new MDContext(LogUtils::removeWorkflowAndTaskInstanceIdMDC);
    }
```

so we can use the `LogUtils` like this:


```java
try (final LogUtils.MDContext ignored = LogUtils.setWorkflowAndTaskInstanceIDMDC(taskRequest.getProcessInstanceId(), taskRequest.getTaskInstanceId())) {
  if (ProcessUtils.kill(taskRequest)) {
    killNumber++;
  }
}
```

Reply:
Good suggestion ❤️